components/match2/commons/match_group_input_util.lua

```diff
@@ -1085,6 +1085,42 @@ function MatchGroupInputUtil.getCharacterName(alias, character)
 	return (assert(alias[character:lower()], 'Invalid character:' .. character))
 end
 
+---@param players {name: string?}[]?
+---@param playerInput string?
+---@param playerLink string?
+---@param options {pagifyPlayerNames: boolean?}?
+---@return integer?
+function MatchGroupInputUtil.findPlayerId(players, playerInput, playerLink, options)
+	if not players then
+		return
+	end
+
+	if not Logic.isEmpty(playerInput) then
+		return
+	end
+	---@cast playerInput -nil
+
+	if not playerLink then
+		playerLink = mw.ext.TeamLiquidIntegration.resolve_redirect(playerInput)
+	end
+	if (options or {}).pagifyPlayerNames then
+		playerLink = Page.pageifyLink(playerLink) --[[@as string]]
+	end
```

Comment:
```suggestion
	if (options or {}).pagifyPlayerNames then
		playerLink = Page.pageifyLink(playerLink or playerInput) --[[@as string]]
	elseif not playerLink then
		playerLink = mw.ext.TeamLiquidIntegration.resolve_redirect(playerInput)
	end
```

saves 1 resolve_redirect lookup (Page.pageifyLink already does it)

Reply:
I think I'd like to keep it as is, makes the coder easier to read and follow imo